app/models/manageiq/providers/redhat/infra_manager/refresh/strategies/api4.rb

```diff
@@ -1,4 +1,59 @@
 module ManageIQ::Providers::Redhat::InfraManager::Refresh::Strategies
   class Api4 < ManageIQ::Providers::Redhat::InfraManager::Refresh::Refresher
+    attr_reader :ems
+
+    def inventory_from_rhv(ems)
+      @ems = ems
+      InventoryWrapper.new(old_inventory: super, ems: ems)
+    end
+
+    class InventoryWrapper
+      attr_reader :old_inventory
```

Comment:
My strategy is to change the code one bit at a time bit to keep it working all the time.
This wrapper will not be there when everything will work through the sdk.